Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -427,6 +426,13 @@ void UPSWorldSubsystem::ResetSaveGameData()
 	{
 		return;
 	}
+
+	if (!ensureMsgf(PSCurrentSpotComponentInternal, TEXT("ASSERT: [%i] %hs:\n'PSCurrentSpotComponentInternal' is null!"), __LINE__, __FUNCTION__))
+	{
+		return;
+	}
+	SetCurrentRowByTag(PSCurrentSpotComponentInternal->GetMeshChecked().GetPlayerTag());
+	PSCurrentSpotComponentInternal->ChangeSpotVisibilityStatus();
```

Comment:
All next functions repeat in both cases and they are required to properly update the progression:
```
	PSCurrentSpotComponentInternal->ChangeSpotVisibilityStatus();
	PSHUDComponent->UpdateProgressionWidgetForPlayer();
	UpdateProgressionActorsForSpot();
```

Can you reuse them in a single place? For instance:
A) introduce new more general function to update progression
B) or update existing one to include call to others, like if `UpdateProgressionActorsForSpot` would call inside `PSCurrentSpotComponentInternal->ChangeSpotVisibilityStatus()` and `PSHUDComponent->UpdateProgressionWidgetForPlayer();`

Reply:
introduce new method: https://github.com/h4rdmol/ProgressionSystem/pull/35/commits/6604821eb3c9ab4fb68e8557f560fa78cb707124